lib/api/core/plugins/pluginsManager.js

```diff
@@ -436,35 +585,49 @@ function initPipes (manager, plugin, pipeWarnTime, pipeTimeout) {
     if (Array.isArray(fn)) {
       fn
         .filter(target => typeof plugin.object[target] === 'function')
-        .forEach(func => registerPipe(manager, plugin, _warnTime, _timeout, pipe, func));
+        .forEach(func => registerPipe(pluginsManager, plugin, _warnTime, _timeout, pipe, func));
     }
     else if (typeof plugin.object[fn] === 'function') {
-      registerPipe(manager, plugin, _warnTime, _timeout, pipe, fn);
+      registerPipe(pluginsManager, plugin, _warnTime, _timeout, pipe, fn);
     }
   });
 }
 
-function initHooks (plugin, kuzzle) {
+/**
+ * @param {PluginsManager} pluginsManager
+ * @param {object} plugin
+ */
+function initHooks (pluginsManager, plugin) {
   _.forEach(plugin.object.hooks, (fn, event) => {
     if (Array.isArray(fn)) {
       fn
         .filter(target => typeof plugin.object[target] === 'function')
-        .forEach(func => registerHook(kuzzle, plugin, event, func));
+        .forEach(func => registerHook(pluginsManager, plugin, event, func));
     }
     else if (typeof plugin.object[fn] === 'function') {
-      registerHook(kuzzle, plugin, event, fn);
+      registerHook(pluginsManager, plugin, event, fn);
     }
   });
 }
 
-function initControllers (controllers, routes, plugin, pluginName) {
+/**
+ * Init plugin controllers
+ *
+ * @param {PluginsManager} pluginsManager
+ * @param {object} plugin
+ * @returns {boolean}
+ */
+function initControllers (pluginsManager, plugin) {
   const controllerImported = Object.keys(plugin.object.controllers).every(controller => {
-    const description = plugin.object.controllers[controller];
     debug('[%s][%s] starting controller registration', plugin.name, controller);
 
+    const
+      description = plugin.object.controllers[controller],
+      errorControllerPrefix = `[!] [WARNING][Plugin Manager]: Unable to inject controller "${clc.bold(controller)}" from plugin "${clc.bold(plugin.name)}":`;
+
     if (typeof description !== 'object' || description === null || Array.isArray(description)) {
       // eslint-disable-next-line no-console
-      console.error(`[Plugin Manager] Error loading plugin ${pluginName}: incorrect controller description type (expected object): \n${description}`);
+      console.warn(clc.yellow(`${errorControllerPrefix} Incorrect controller description type (expected object, got: "${clc.bold(typeof description)}")`));
       return false;
```

Comment:
should throw a `PluginImplementationError` with the above message, instead of returning `false`